edifice/engine.py

```diff
@@ -1937,6 +1939,8 @@ def use_state(
             hook = hooks[h_index]
 
         def setter(updater):
+            if element._was_deleted:
```

Comment:
Maybe here instead we should do
```suggestion
            if element not in self._hook_state:
```
And then we don't need the extra `_was_deleted` member.